spec/models/capabilities/scopes/default_spec.rb

```diff
@@ -101,117 +90,181 @@
       let(:members) { [member] }
 
-      it_behaves_like 'is empty'
+      include_examples 'is empty'
+
+      context 'with a module being activated with a public permission' do
+        before do
+          project.enabled_module_names = ['activity']
+        end
+
+        include_examples 'consists of contract actions', with: 'the actions of the public permission' do
+          let(:expected) do
+            [
+              ['activities/read', user.id, project.id]
+            ]
+          end
+        end
+      end
     end
 
-    context 'with a global member without a permission' do
+    context 'with a global member without any permissions' do
       let(:members) { [global_member] }
 
-      it_behaves_like 'is empty'
+      include_examples 'is empty'
     end
 
-    context 'with a non member role without a permission' do
+    context 'with a non member role without any permissions' do
       let(:members) { [non_member_role] }
 
-      it_behaves_like 'is empty'
+      include_examples 'is empty'
+
+      context 'with the project being public and having a module activated with a public permission' do
+        before do
+          project.update(public: true)
+          project.enabled_module_names = ['activity']
+        end
+
+        include_examples 'consists of contract actions', with: 'the actions of the public permission' do
+          let(:expected) do
+            [
+              ['activities/read', user.id, project.id]
+            ]
+          end
+        end
+      end
     end
 
-    context 'with a global member with an action permission' do
+    context 'with a global member with a global permission' do
       let(:global_permissions) { %i[manage_user] }
       let(:members) { [global_member] }
 
-      it_behaves_like 'consists of contract actions' do
+      include_examples 'consists of contract actions', with: 'the actions of the global permission' do
         let(:expected) do
           [['users/create', user.id, nil],
            ['users/read', user.id, nil],
            ['users/update', user.id, nil]]
         end
       end
+
+      context 'with the user being locked' do
+        before do
+          user.locked!
+        end
+
+        include_examples 'is empty'
+      end
     end
 
-    context 'with a member with an action permission' do
+    context 'with a member with a project permission' do
       let(:permissions) { %i[manage_members] }
       let(:members) { [member] }
 
-      it_behaves_like 'consists of contract actions' do
+      include_examples 'consists of contract actions', with: 'the actions of the project permission' do
         let(:expected) do
           [['memberships/create', user.id, project.id],
            ['memberships/destroy', user.id, project.id],
            ['memberships/update', user.id, project.id]]
         end
       end
-    end
-
-    context 'with a global member with an action permission and the user being locked' do
```

Comment:
Is this covered somewhere else?

Reply:
Yes. It may not appear clearly: I nested some tests in other existing context to better differentiate the normal case and the special cases. Now it's on line 155.

```
Capabilities::Scopes::Default
  .default
    with a global member with a global permission
      includes the actions of the global permission for the scoped to user
      with the user being locked                                               <==== *here*
        is empty for the scoped to user
```